include/ImmediateContext.hpp

```diff
@@ -222,6 +222,17 @@ class CMultiLevelPool
         }
     }
 
+    void AggressiveTrim(UINT64 CurrentFenceValue)

```

Comment:
I would've just piped the `aggressive` bool into this function too instead of making it a new one. Then the trim threshold just need to be `aggressive ? 2 : m_TrimThreshold`. Then you don't need such a redundant branch in `ImmediateContext::TrimResourcePools` either.